src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -232,14 +233,16 @@ private async Task<bool> HeartbeatAsync(CancellationToken cancellationToken)
             }
             else
             {
-                newDescription = _baseDescription.With(lastUpdateTimestamp: DateTime.UtcNow);
+                newDescription = _baseDescription.With("Heartbeat", lastUpdateTimestamp: DateTime.UtcNow);
             }
 
             if (heartbeatException != null)
             {
-                newDescription = newDescription.With(heartbeatException: heartbeatException);
+                newDescription = newDescription.With("Heartbeat", heartbeatException: heartbeatException);
             }
 
+            newDescription = newDescription.With("Heartbeat", lastHeartbeatTimestamp: DateTime.UtcNow);
```

Comment:
can we add a reason only once here? It looks like no reason to specify the same keys several times?

Reply:
All other `With` arguments have explicit adding the argument name before the value. If I'm not mistaken, there is no harm to add it here too

Reply:
Sure... originally reasonChanged was a mandatory parameter which is why it was included in every call to With.

Reply:
done